### ISSUE TITLE
Installation instructions: Multiple worrying npm vunerability warnings

### ISSUE THREAD

**Issue by TimFrance** [adaptlearning/adapt_authoring#2419]
## Expected Behaviour
There was no indication that I could/should ignore the many npm vunerability warnings (I found that out by searching the forums). If they really are not important you should tell users to ignore them, suppress them, or point to solutions if they need to be acted on.

## Actual Behaviour
>npm audit
                       === npm audit security report ===
# Run  npm install migrate-mongoose@4.0.0  to resolve 1 vulnerability
SEMVER WARNING: Recommended action is a potentially breaking change
  Low             Regular Expression Denial of Service
  Package         braces
  Dependency of   migrate-mongoose
  Path            migrate-mongoose > babel-cli > chokidar > anymatch >
                  micromatch > braces
  More info       https://npmjs.com/advisories/786
                                 Manual Review
             Some vulnerabilities require your attention to resolve
          Visit https://go.npm.me/audit-guide for additional guidance
  High            Prototype Pollution
  Package         lodash
  Patched in      >=4.17.11
  Dependency of   grunt-merge-json
  Path            grunt-merge-json > lodash
  More info       https://npmjs.com/advisories/782
  High            Prototype Pollution
  Package         lodash
  Patched in      >=4.17.12
  Dependency of   grunt-merge-json
  Path            grunt-merge-json > lodash
  More info       https://npmjs.com/advisories/1065
  High            Arbitrary File Overwrite
  Package         fstream
  Patched in      >=1.0.12
  Dependency of   unzip
  Path            unzip > fstream
  More info       https://npmjs.com/advisories/886
found 4 vulnerabilities (1 low, 3 high) in 7369 scanned packages
  1 vulnerability requires semver-major dependency updates.
  3 vulnerabilities require manual review. See the full report for details.

## Steps to Reproduce
1. Just install on Windows 10, following the instructions to the letter and installing all dependencies from scratch. (downloaded today)
1.
1.

## Versions
  - Authoring Tool Version: 0.9.0
  - Framework Version: ?? Downloaded today
  - Node.js Version: v10.16.3
  - MongoDB Version: Downloaded today
  - Operating System: Windows 10
  - Browser: Chome Version 76.0.3809.100 (Official Build) (64-bit)

**Comment by oliverfoster** [comment]
I'm glad you found an answer, the JavaScript world can be a tad expansive sometimes. 

The above output is generated by the npm cli which is bundled with NodeJS. npm isn't a part of the infrastructure we have control over and does have a tendency to change regularly. 

The warnings arise from our use of third party modules that come with their own dependencies, because of that we have no control over a large part of the foundational environment on which we build the AAT (thankfully, it would've taken decades to build).

As you have found the warnings can be ignored and ignoring them is best done with a bit of research and some understanding about how these exploits are executed in the wild.

Do you have any suggestion as to how you'd like this issue solved?

**Comment by TimFrance** [comment]
Hi Oliver

I am not technical enough to understand how to interpret the gravity of
those warnings, I was only interested in getting access to the tool.

Although installation finished successfully, it took a long time and I was
uneasy about the warnings that I could not control.

I thought that by poking around and finding where things might be going
wrong, I could make the task easier for someone somewhere to correct. It is
the only way that I can contribute.

Any I understand your answer in that case perhaps the "solution" would just
information people in the instructions that they will get some warnings and
it is safe to ignore them. (provided that all the dependencies git/ffmpeg
etc) are up to date

On Windows 10 the current versions of these dependencies are:


  - Node.js Version: v10.16.3

  - MongoDB Compass Community Version: 1.19.6

  - ffmpeg version 4.2

  - grunt-cli v1.3.2
  - grunt v1.0.4

  - npm version "6.9.0"

  - git version 2.23.0.windows.1
That would encourage newbies like me to install more easily.

Thanks for your help, and sorry about posting similar problems 3 times, I
was only trying to help.

The good news is that despite my lack of knowledge I managed to install
Adapt, and it seems to work fine. Now I can create some great content.

Tim

**Comment by oliverfoster** [comment]
Cool, I'll see if I can get someone to add a message about the npm security vulnerabilities to the installation instructions. 

Did you find any particularly useful references which helped you understand what they are?

**Comment by oliverfoster** [comment]
You might find it easier to reply on github rather than through email. I have a tendency to edit my answers (as my spelling can be terrible) - which I don't think is reflected in the email chain.

**Comment by TimFrance** [comment]
As in my posts, I just compared the headings of the package.json file with
the recommendations at the link to npmjs.com given in the warnings and
found packages that were out of date, sometimes different versions of the
same package in different branches.

e.g.for

*High            Arbitrary File Overwrite*

  Package         fstream

  Patched in      >=1.0.12

  Dependency of   unzip

  Path            unzip > fstream

  More info       https://npmjs.com/advisories/886

and found that the two fstream packages in the installation were older than
1.0.12. On the other hand, I had no way to evaluate if that meant the
installation was OK, dangerous, or would not work.

Thanks for your support

Tim

**Comment by oliverfoster** [comment]
I meant more: How did you figure out that you are able to ignore the vulnerability warnings on your own, was their anything you read which helped you understand their context better?

**Comment by TimFrance** [comment]
Not really. I searched and found one comment on a similar problem, and the
advice was to ignore it. I cannot find the page in my history, but it
wasn't much.

I just decided I had no choice and ploughed on.

I felt lucky that it worked.

Tim

**Comment by TimFrance** [comment]
Hi Oliver
Looking around Github and Youtube there seems to be very little (recent)
activity around ADAPT, on the other hand, you have been very quick to reply
and very helpful.

I am finding ADAPT to pretty good, even though there are a few bugs. As far
as I have found there are not many tutorials, or at least not enough to
encourage a wider population to use it.

So I am wondering if this project is still alive and progressing, is it
dead, or could it be woken up?

Best regards

Tim Baker (France)

**Comment by oliverfoster** [comment]
Come talk to us in https://gitter.im/adaptlearning/adapt_framework
